invenio_rdm_records/resources/serializers/schemas.py

```diff
@@ -79,3 +86,17 @@ def get_contributors(self, obj):
         return [
             c["person_or_org"]["name"] for c in obj["metadata"].get("contributors", [])
         ] or missing
+
+    @post_dump(pass_original=True)
+    def post_dump_pipeline(self, data, original, many, **kwargs):
+        """Defines sequence of post dump steps."""
+        result = deepcopy(data)
+        for processor in self.processors:
+            if not hasattr(processor, "post_dump"):
+                raise TypeError("Post process processors must be callables")
+            # TODO does it need to be instantiated?
+            dumped_data = processor().post_dump(result, original)
+            merged = _merge_dictionaries(result, dumped_data)
```

Comment:
This is a bit tricky, since we're applying steps in sequence to an already serialized record. 
E.g. order matters, different steps might try to "override" something that was already computed. 
For now, `_merge_dictionaries` only accepts new keys or keys whose value is a list. For the latter, values are simply appended.

Reply:
I would avoid this merging logic by assuming that the processors receive the full object and are responsible for resolving order/key conflicts in the data.